### ISSUE TITLE
RoutesInfo support for query string

### ISSUE THREAD

**Issue by quoininc-huudatran** [Quoin/react-utils#18]
It needs to be able to handle query params.

```javascript
{
  name: ROUTES.SEARCH,
  path: '/search{?q}',
}
```

**Comment by quoininc-huudatran** [comment]
It should also handle the case of:
```
{
  name: ROUTES.PARENT,
  path: '/parent{?q}',
  routes: [{
    name: ROUTES.CHILD,
    path: '/child{?p}
  }],
}
```

this should end up with `/parent/child{?p}`.

**Comment by quoininc-huudatran** [comment]
Fixed by 0594c27
